test/end2end_test.go

```diff
@@ -3741,13 +3741,9 @@ func (s) TestClientStreaming_ReturnErrorAfterSendAndClose(t *testing.T) {
 	}
 }
 
-// Tests the behavior for server-side streaming RPCs when client calls SendMsg twice.
-// The first client.SendMsg() sends EOF along with the message. When client calls a
-// second SendMsg, it triggers a RST_STREAM which cancels the stream context on the
-// server. There would be a race, the RST_STREAM could cause the server’s first RecvMsg
-// to fail, even if the request message was already delivered. By synchronizing, we
-// ensure that the server has read the first message before the client triggers RST_STREAM
-// and validating expected error codes.
+// Tests the behavior for server-side streaming when client calls SendMsg twice.
+// Second call to SendMsg should fail with Internal error and result in closing
+// the connection with a RST_STREAM.
 func (s) TestServerStreaming_ClientCallSendMsgTwice(t *testing.T) {
 	// To ensure server.recvMsg() is successfully completed.
```

Comment:
What I was hoping for was more like this:

```suggestion
	// To ensure server.recvMsg() is successfully completed.  Otherwise, if the client application
	// attempts to send a second request message, that will trigger a RST_STREAM from the
	// client due to the application violating the RPC's protocol.  The RST_STREAM will prevent
	// the method handler from being called.
```

Reply:
Done.